oci/layout/oci_transport.go

```diff
@@ -90,7 +99,26 @@ func NewReference(dir, image string) (types.ImageReference, error) {
 		return nil, err
 	}
 
-	return ociReference{dir: dir, resolvedDir: resolved, image: image}, nil
+	if sourceIndex != -1 && sourceIndex < 0 {
+		return nil, fmt.Errorf("Invalid oci: layout reference: index @%d must not be negative", sourceIndex)
+	}
+	if sourceIndex != -1 && image != "" { // Coverage: New*Reference do not pass down that condition
```

Comment:
Please move that comment to the uncovered `getManifestDescriptor` (line 222).

We do actually test _this_ line, so it does not need a comment justifying missing test coverage.

Reply:
https://github.com/containers/image/pull/2673